docs/DeveloperGuide.md

```diff
@@ -1,29 +1,87 @@
+
 # Developer Guide
 
 ## Acknowledgements
+<!-- @@author kaixiangg -->
+[Java Standard Library](https://docs.oracle.com/javase/8/docs/api/): Utilized extensively for collections, date
+handling, and file operations.
+
+[ANSI Escape Codes](https://en.wikipedia.org/wiki/ANSI_escape_code): Adapted for colored console outputs in Ui.java.
+
+[JUnit 5](https://junit.org/junit5/docs/current/user-guide/): Used for unit testing.
+<!-- @@author  -->
+# Design & Implementation
+
+## Architecture
+
+The application follows a modular design:
+
+* **FinTrack (Entry Point)**: Initializes components and manages the application loop.
+* **Parser**: Handles user inputs and ensures validation.
+* **Commands**: Manages mapping and execution of commands.
+* **Expense & ExpenseList**: Data models representing expenses and collections of expenses.
+* **Ui**: Provides a consistent user interface through console outputs.
+
+
+
+<!-- @@author kaixiangg -->
+### CRUD Operations Implementation
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+#### Overview
 
-## Design & implementation
+The CRUD (Create, Read, Update, Delete) operations
+are central to FinTrack's expense management functionality.
+They are implemented through the `Commands` class which acts as
+a controller, coordinating between the user interface,
+data parsing, and data storage components.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The CRUD operations follow a layered architecture:
 
+1. User Interface Layer (`Ui` class)
+2. Control Layer (`Commands` class)
+3. Logic Layer (`Parser` and `Categories` classes)
+4. Data Layer (`ExpenseList` and `Storage` classes
+
+![CRUD_UML.png](CRUD_UML.png)
+
+## Classes and Their Responsibilities
+
+* **Commands**:
+    * Maps user commands (`add`, `viewmonth`, `history`, `update`, `delete`, `budget`, `recurring`, `category add`,
```

Comment:
Perhaps it will be great to include sequence diagram for other scenarios such as `category add` and `recurring`. 
It will also be good to have description following each sequence diagram to explain the logic flow taken in detail and elaborate on considerations.